Mathlib/LinearAlgebra/QuadraticForm/Basis.lean

```diff
@@ -0,0 +1,63 @@
+/-
+Copyright (c) 2024 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.LinearAlgebra.QuadraticForm.Basic
+
+/-!
+# Constructing a bilinear map from a quadratic map, given a basis
+
+This file provides an alternative to `QuadraticMap.associated`; unlike that definition, this one
+does not require `Invertible (2 : R)`. Unlike that definition, this only works in the presence of
+a basis.
+-/
+
+namespace QuadraticMap
+
+variable {ι R M N} [LinearOrder ι]
+variable [CommRing R] [AddCommGroup M] [AddCommGroup N] [Module R M] [Module R N]
+
+/-- Given an ordered basis, produce a bilinear form associated with the quadratic form.
+
+Unlike `QuadraticMap.associated`, this is not symmetric; however, as a result it can be used even
+in characteristic two. When considered as a matrix, the form is triangular. -/
+noncomputable def toBilin (Q : QuadraticMap R M N) (bm : Basis ι R M) : LinearMap.BilinMap R M N :=
```

Comment:
(I think this is what I was trying to do here: https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Is.20Finset.20totally.20ordered.3F )

Reply:
I'm surprised I didn't have anything more interesting to say at the time